sources/iTermSemanticHistoryController.m

```diff
@@ -73,9 +87,13 @@ - (NSString *)getFullPath:(NSString *)path
         return nil;
     }
 
-    // If it's in parens, strip them.
-    if (path.length > 2 && [path characterAtIndex:0] == '(' && [path hasSuffix:@")"]) {
-        path = [path substringWithRange:NSMakeRange(1, path.length - 2)];
+    // If it's in any form of bracketed delimiters, strip them
```

Comment:
I think this implementation chooses performance over clarity, which isn't the right tradeoff here (in the midst of filesystem access). I'd suggest doing something like this:

    while ([path isBracketed]) {
        path = [path stringByRemovingBrackets];
    }

and add -isBracketed and -stringByRemovingBrackets methods to NSStringIterm, like:

    - (BOOL)isBracketed {
        if (self.length < 2) return NO;
        if ([self hasPrefix:@"("] && [self hasSuffix:@")"]) return YES;
        if ([self hasPrefix:@"<"] && [self hasSuffix:@">"]) return YES;
        etc.
        return NO;
    }